lib/src/tes/mod.rs

```diff
@@ -0,0 +1,328 @@
+    ) -> Result<TesListTasksResponse, Box<dyn std::error::Error>> {
+        let params_value = params.map(|p| {
+            let mut map = serde_json::Map::new();
+            
+            if let Some(name_prefix) = p.name_prefix {
+                map.insert("name_prefix".to_string(), json!(name_prefix));
+            }
+            if let Some(state) = p.state {
+                map.insert("state".to_string(), json!(state));
+            }
+            if let Some(tag_key) = p.tag_key {
+                map.insert("tag_key".to_string(), json!(tag_key));
+            }
+            if let Some(tag_value) = p.tag_value {
+                map.insert("tag_value".to_string(), json!(tag_value));
+            }
+            if let Some(page_size) = p.page_size {
+                map.insert("page_size".to_string(), json!(page_size));
+            }
+            if let Some(page_token) = p.page_token {
+                map.insert("page_token".to_string(), json!(page_token));
+            }
+            if let Some(view) = p.view {
+                map.insert("view".to_string(), json!(view));
+            }
+
+            json!(map)
+        });
+        // println!("{:?}",params_value);
+        // Make the request with or without parameters based on the presence of params
+        let response = if let Some(params_value) = params_value {
+            self.transport.get("/tasks", Some(params_value)).await
+        } else {
+            self.transport.get("/tasks", None).await
+        };
+
+        match response {
+            Ok(resp_str) => {
+                let task: TesListTasksResponse = from_str(&resp_str)?;
+                Ok(task)
+            }
+            Err(e) => {
+                eprintln!("HTTP request failed: {:?}", e);
+                Err(Box::new(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("HTTP request failed: {:?}", e),
+                )))
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::configuration::Configuration;
+    use crate::tes::models::TesTask;
+    use crate::tes::ListTasksParams;
+    use crate::tes::Task;
+    use crate::tes::TesState;
+    use crate::tes::TES;
+    use crate::test_utils::{ensure_funnel_running, setup};
+    // use crate::tes::models::TesCreateTaskResponse;
+
+    async fn create_task() -> Result<(Task, TES), Box<dyn std::error::Error>> {
+        // setup(); – should be run once in the test function
+        let mut config = Configuration::default();
+        let funnel_url = ensure_funnel_running().await;
+        config.set_base_path(&funnel_url);
+        let tes = match TES::new(&config).await {
+            Ok(tes) => tes,
+            Err(e) => {
+                println!("Error creating TES instance: {:?}", e);
+                return Err(e);
+            }
+        };
+        let file_path = std::env::var("TASK_FILE_PATH").unwrap_or_else(|_| "./lib/sample/grape.tes".to_string());
```

Comment:
Okay, seems I don't understand what `grape.tes` is for. Because I am surprised to find that in the models...?

Reply:
This grape.tes is in the tests module only, and is _only being used for tesing_. Currently, this is being used to make a sample task. It is in a function, since this file is being used in multiple tests, so this way redundancy can be avoided. 
To be noted, that this now replaced by sample.tes.

Reply:
@uniqueg `grape.tes` is in the `sample` dir, not models, it contains TES Task definition JSON data, which was quickly taken from the [pre-initial CLI implementation](https://github.com/genxnetwork/ga4gh-cli). as agreed in another conversation here, [will be updated to a more generic one and moved](https://github.com/elixir-cloud-aai/ga4gh-sdk/pull/13#discussion_r1665281867) to ./tests/files